### PR TITLE
[GPU] Refactor Comparison, Conversion, ConvertColorI420

### PR DESCRIPTION
**Pull request by andrei-cv**
### Details:
 - *Comparison, Conversion, ConvertColorI420*

### Tickets:
 - *CVS-111379*

### CODE REVIEW

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/convert_color_i420.cpp [hunk 1]
```diff
@@ -4,42 +4,76 @@
...
 INSTANTIATE_TEST_SUITE_P(smoke_TestsConvertColorI420_acc,
-                         ConvertColorI420AccuracyTest,
-                         ::testing::Combine(::testing::Values(ov::Shape{1, 16 * 6, 16, 1}),
+                         ConvertColorI420LayerTest,
+                         ::testing::Combine(::testing::ValuesIn(ov::test::static_shapes_to_test_representation(
+                                                generate_input_static_shapes({{1, 16 * 6, 16, 1}}, true))),
                                             ::testing::Values(ov::element::u8),
-                                            ::testing::Bool(),
-                                            ::testing::Bool(),
+                                            ::testing::Values(false),
```
Comment (Lyamin-Roman) [inline src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/convert_color_i420.cpp:68]: Why this test only for single plane and BGR? All cases have been checked before

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/convert_color_i420.cpp [hunk 2]
```diff
@@ -4,42 +4,76 @@
...
+            one_result_shapes.push_back(uvShape);
+        }
+        result_shapes.push_back(one_result_shapes);
+    }
+    return result_shapes;
 };
 
-INSTANTIATE_TEST_SUITE_P(smoke_TestsConvertColorI420,
+auto in_shapes_single_plain_static     = generate_input_static_shapes(inShapes_nhwc, true);
+auto in_shapes_not_single_plain_static = generate_input_static_shapes(inShapes_nhwc, false);
```
Comment (Lyamin-Roman) [inline src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/convert_color_i420.cpp:43]: nit: not_single_plain -> three_planes

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/comparison.cpp [hunk 3]
```diff
@@ -63,20 +74,17 @@ const std::vector<InputShapesTuple> inputShapesIsOps = {
...
-                                            ::testing::ValuesIn(inputsPrecisionsIsOps),
-                                            ::testing::Values(ngraph::helpers::ComparisonTypes::IS_FINITE,
-                                                              ngraph::helpers::ComparisonTypes::IS_NAN),
-                                            ::testing::Values(ngraph::helpers::InputLayerType::CONSTANT),
-                                            ::testing::Values(InferenceEngine::Precision::UNSPECIFIED),
-                                            ::testing::Values(InferenceEngine::Precision::BOOL),
+                         ::testing::Combine(::testing::ValuesIn(ov::test::static_shapes_to_test_representation(inputShapesIsOps)),
+                                            ::testing::ValuesIn(comparisonOpTypesIs),
+                                            ::testing::Values(ov::test::utils::InputLayerType::CONSTANT),
+                                            ::testing::Values(ov::element::f32),
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/comparison.cpp:87]: Please add `ov::element::f16` data type as well, as it was before

Comment (sshlyapn) [reply src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/comparison.cpp]: @andrei-cv why did you remove f32 data type? Let's test them both

**Review by sshlyapn — APPROVED** [review]

**Review by Lyamin-Roman — APPROVED** [review]

**Pull request merged**